doc/en/user/source/security/csp.rst

```diff
@@ -0,0 +1,270 @@
+or by dragging and dropping the rule's row in the table.
+
+.. figure:: images/csp-rule.png
+
+   CSP Rule Configuration
+
+* A name that is unique among the rules within the specific policy must be provided in the
+  :guilabel:`Name` text field when adding a new rule.
+* The :guilabel:`Description` text field provides an optional description to help administrators
+  understand what the rule does.
+* The :guilabel:`Enabled` checkbox will enable/disable the rule.
+* The :guilabel:`Request Filter` text field contains the filter to apply to each user request to
+  determine whether to add this rule's directives to the CSP header value. (see
+  :ref:`security_csp_filters` below)
+* The :guilabel:`Header Directives` text field contains the CSP directives to add to the header
+  value when a request matches this rule's filter. (see :ref:`security_csp_directives` below)
+
+.. note::
+    After saving a rule, make sure to save the policy and then save/apply the configuration.
+
+.. _security_csp_filters:
+
+Request Filters
+---------------
+
+The filter contains a string of predicates concatenated with the string ``AND`` and the rule's
+directives will be applied to a request only if all of the predicates match the request. There
+are three types of predicates that can be used:
+
+* PATH(regex): Returns true if the URL-decoded request path matches the regular expression. The
+  regex will be tested against the path that is relative to GeoServer's context root and starting
+  with a forward slash.
+  Example: ``PATH(^/([^/]+/){0,2}wms/?$)``
+* PARAM(key_regex,value_regex): Returns true if all query parameters with a URL-decoded key that
+  match the key_regex have a URL-decoded value that match the value_regex. The value regex will be
+  tested against an empty string if no query parameters matched the key regex.
+  Example: ``PARAM((?i)^service$,(?i)^wms$)``
+* PROP(key,value_regex): Returns true if the value for the property key matches the regex. The key
+  is case-sensitive and must contain must contain the string ``GeoServer``, ``GeoTools``, or
+  ``GeoWebCache`` anywhere in the key (case-insensitive). The regex will be tested against an empty
+  string if the property is not set. This is primarily intended for the default configuration and
+  may not be useful to administrators.
+  Example: ``PROP(GEOSERVER_CONSOLE_DISABLED,(?i)^(?!true$).*$)``
+
+.. note::
+    The ``(?i)`` at the beginning of the regular expression will use case-insensitive matching and
+    enclosing the pattern inside of the ``^$`` characters will match the entire string. See the
+    `Regular Expressions Tutorial <https://docs.oracle.com/javase/tutorial/essential/regex/>`_ for
+    more information about how to use Java regular expressions.
+
+Leaving the filter blank will cause this rule to match all requests and should only be used on the
+last rule in a policy since any additional rules would never be checked.
```

Comment:
That seems really hard to understand; if blank is aways used then it does not need to be shown (or maybe indicated another way).